update description about dumpling privileges (#11326)
pingcap/docs: dumpling-overview.md

### Required privileges

- SELECT: Required when exporting tables.

> - RELOAD: Required when using `consistency flush`. RDS databases do not support this privilege. Therefore, you can ignore this privilege if the upstream is an RDS database or a managed service.

Reviewer: RDS （exclude TiDB ）

> - RELOAD: Required when using `consistency flush`. RDS databases do not support this privilege. Therefore, you can ignore this privilege if the upstream is an RDS database or a managed service.

Author: ```suggestion
- RELOAD: Required when using `consistency flush`. Note that only TiDB supports this privilege. When the upstream is an RDS database or a managed service, you can ignore this privilege.
```

- LOCK TABLES: Required when using `consistency lock`. This privilege must be granted for all the databases and tables to be exported.

> - REPLICATION CLIENT: Required when exporting metadata to record data snapshot. This privilege is optional.

Reviewer 2: ```suggestion
- REPLICATION CLIENT: Required when exporting metadata to record data snapshot. This privilege is optional if you do not need to export metadata.
```

Author: IMO, optional is a fixed attribute and does not change if the user wants to export metadata. How about:
> This privilege is optional and you can ignore it if you do not need to export metadata.

Reviewer 2: LGTM
